e2e-tests/src/test/adder.rs

```diff
@@ -13,14 +18,31 @@ pub async fn adder() -> Result<()> {
     let config = setup_test();
 
     let (conn, _authority, account) = basic_test_context(config).await?;
-    let contract = AdderInstance::new(
+
+    let contract = Arc::new(AdderInstance::new(
         &config.test_case_params.adder,
         &config.test_case_params.adder_metadata,
-    )?;
+    )?);
+
+    let listen_conn = conn.clone();
+    let listen_contract = contract.clone();
+    let (tx, mut rx) = unbounded();
+    let listen = || async move {
+        listen_contract_events(&listen_conn, &[listen_contract.as_ref().into()], tx).await?;
+        <Result<(), anyhow::Error>>::Ok(())
```

Comment:
Do we need it? `listen_contract_events` already returns `Result<(), anyhow::Error>` so we should be OK if we drop the `;`

Reply:
An `|| async move` has problems with its own identity and so the compiler won't be able to deduce the type in that case. See https://rust-lang.github.io/async-book/07_workarounds/02_err_in_async_blocks.html

Reply:
Bluh, I didn't know that. thanks.